web/modules/custom/contact_information/contact_information.module

```diff
@@ -112,29 +112,50 @@ function contact_information_entity_auto_label(EntityInterface $entity) {
-  if ($entity->getEntityTypeId() != 'rdf_entity') {
+  if ($entity->bundle() !== 'contact_information') {
     return AccessResult::neutral();
   }
-  if ($entity->bundle() != 'contact_information') {
-    return AccessResult::neutral();
-  }
-  // Only revoke access for anonymous users.
-  if ($account->isAuthenticated()) {
-    return AccessResult::neutral();
+
+  // Inline entity form module is using the 'update' as a name for the edit
+  // operation as it is with all entities apart from nodes that retained the
+  // 'edit' permission for legacy reasons. However, rdf entities also use this
+  // naming convention. Rename 'update' to 'edit' when inline_entity_form checks
+  // for the update permission.
+  if ($operation === 'update') {
+    $operation = 'edit';
   }
-  // Get entity from route.
-  $route = \Drupal::routeMatch();
-  $route_name = $route->getRouteName();
-  if ($route_name != 'entity.rdf_entity.canonical') {
-    return AccessResult::neutral();
+
+  // For anonymous users, deny access to the canonical route.
+  if ($account->isAnonymous()) {
+    $route_name = \Drupal::routeMatch()->getRouteName();
+    if ($route_name === 'entity.rdf_entity.canonical') {
+      return AccessResult::forbidden('Anonymous users should not see contact information on the canonical route.');
+    }
   }
-  $route_entity = $route->getParameter('rdf_entity');
-  if (empty($route_entity)) {
-    return AccessResult::neutral();
+  elseif ($operation == 'edit') {
+    $allowed_transitions = \Drupal::service('joinup_core.workflow.helper')->getAvailableTargetStates($entity, $account);
+    return empty($allowed_transitions) ? AccessResult::forbidden() : AccessResult::allowed();
   }
-  if ($entity->id() == $route_entity->id()) {
-    return AccessResult::forbidden('Anonymous users should not see contact information on the canonical route.');
+  // Use an elseif to avoid checking memberships for anonymous users.
+  elseif ($operation === 'delete') {
+    // The contact information is not a group content rather than it is simply
+    // referenced by a collection or a solution.
+    // To grant access to group owner/facilitators, we need to identify first
+    // which groups reference the contact information.
+    /** @var \Drupal\joinup_core\JoinupRelationManagerInterface $relation_manager */
+    $relation_manager = \Drupal::service('joinup_core.relations_manager');
+    if ($groups = $relation_manager->getContactInformationRelatedGroups($entity)) {
+      /** @var \Drupal\og\MembershipManagerInterface $membership_manager */
+      $membership_manager = \Drupal::service('og.membership_manager');
+      foreach ($groups as $group) {
+        /** @var \Drupal\og\Entity\OgMembership $membership */
+        $membership = $membership_manager->getMembership($group, $account->id());
+        if ($membership && $membership->hasPermission("delete contact_information rdf entity")) {
+          return AccessResult::allowed();
+        }
+      }
```

Comment:
But, this means that if a user has access to delete a contact information from a single collection or solution, then it can delete the contact information for all the groups that reference it?

What happens in this case? Is the contact information removed for all groups?

Reply:
I have raised this question to the Functional Contractor. They will decide what happens in this case. Probably they should not be able to delete these entities. This should be available only to moderators. However, this refers only to very old contact information entities. It seems that we have a small number (36) that belong to multiple groups. It will indeed be weird to have them deleted even by accident, let alone have this break the website.

Reply:
OK so the new expected behavior as defined by the functional team is that we will only consider the first collection or solution that was associated with a contact information, and ignore any subsequent ones. There are a small number of old collections and solutions that will be affected if their contact information is deleted, but it is decided that the risk is very small and can be rectified manually by moderators by creating new contact information, if this issue is raised through support channels.

I personally would have preferred a solution that safeguards the data integrity without requiring manual intervention but I respect the decision of the functional team, and can confirm this is now implemented according to the requested specification.